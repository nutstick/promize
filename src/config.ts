/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

/* eslint-disable max-len */

if (process.env.BROWSER) {
  throw new Error(
    'Do not import `config.ts` from inside the client-side code.',
  );
}

export const locales = [
  /* @intl-code-template '${lang}-${COUNTRY}', */
  'en-US',
  'th-TH',
  /* @intl-code-template-end */
];

export const port = process.env.PORT || 3000;

export const mongodb = {
  host: process.env.MONGO_HOST || `localhost`,
  port: parseInt(process.env.MONGO_PORT, 10) || 27017,
  database: process.env.MONGO_DB || 'promize',
};

export const api = {
  // API URL to be used in the client-side code
  clientUrl: process.env.API_CLIENT_URL || '',
  // API URL to be used in the server-side code
  serverUrl:
    process.env.API_SERVER_URL ||
    `http://localhost:${process.env.PORT || 3000}`,
};

export const databaseUrl = process.env.DATABASE_URL || 'sqlite:database.sqlite';

// Web analytics
export const analytics = {
  // https://analytics.google.com/
  googleTrackingId: process.env.GOOGLE_TRACKING_ID, // UA-XXXXX-X
};

// Authentication
export const auth = {
  jwt: { secret: process.env.JWT_SECRET || 'Typescript React Starter Kit' },

  // https://developers.facebook.com/
  // ID: promise.seproject
  // Password: promise123
  facebook: {
    id: process.env.FACEBOOK_APP_ID || '1946061138994426',
    secret:
      process.env.FACEBOOK_APP_SECRET || 'f662d8cdf0bc78f455db8f45eeda6a81',
  },

  // https://cloud.google.com/console/project
  google: {
    id:
      process.env.GOOGLE_CLIENT_ID ||
      '251410730550-ahcg0ou5mgfhl8hlui1urru7jn5s12km.apps.googleusercontent.com',
    secret: process.env.GOOGLE_CLIENT_SECRET || 'Y8yR9yZAhm9jQ8FKAL8QIEcd',
  },

  // https://apps.twitter.com/
  twitter: {
    key: process.env.TWITTER_CONSUMER_KEY || 'Ie20AZvLJI2lQD5Dsgxgjauns',
    secret:
      process.env.TWITTER_CONSUMER_SECRET ||
      'KTZ6cxoKnEakQCeSpZlaUCJWGAlTEBJj0y2EMkUBujA7zWSvaQ',
  },
};

export const uploadDir = './public/images';
