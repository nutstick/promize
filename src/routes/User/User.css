@import '../../components/variables.css';

.root {
  padding: 0.6em 0 40px;
  margin-top: -120px;
}

.container {
  display: flex;
  min-height: 100%;
}

.sidebar {
  // flex: 0 0 12em;
  flex: 1 0 auto;
  display: flex;
  order: -1;
  flex-direction: column;
  
}

.profile {
  padding: 1.5rem 1.5rem 0;
}

.profileImage{
  box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
  min-height: 120px;
}

.name {
  padding-top: 1rem;
  text-align:center;
  font-size: larger;
  font-weight: bold;
  line-height: 2rem;
}

.detail {
  color: #5a5a5a;
  text-align:center;
  line-height: 1.5rem;
}

.menu {
  // padding: 0 1.5rem 1.5rem 1.5rem;
}

.mainContent{
  flex: 1;
}

.container {
  margin: 30px;
}

.content {
  background-color: rgba(0, 0, 0, 0.01);
}
.active {
  font-weight: bold;
  color: rgb(231, 53, 53);
}

li {
  width: 100%;
  margin: 0;
  display: block;
  padding: 1rem;
  padding-left: 2em;
  border: 1px solid #e4e0e0;
  border-top-color: transparent;
  border-left-color: transparent;
  border-right-color: transparent;
}

li:hover {
  color: rgb(231, 53, 53);
}

a:hover {
  color: rgb(231, 53, 53);
}

.menuList {
  list-style-type: none;
  display:inline-block;
  margin: 0;
  padding: 0;
  width: 100%;
}

@media (max-width: 620px) {
  .mainContent {
     flex-direction: column;
  }
  ul {
    list-style-type: none;
    display: flex;
    margin: 0;
    padding: 0;
  }

  li {
    margin: 0; 
    float:left;
    padding: 1rem;
    cursor: pointer;
    text-align: center;
    border: 1px solid #e4e0e0;
    border-bottom-color: transparent;
  }
}
@media (min-width: 620px) {
  .iconNewLine {
    display: none;
  }
}