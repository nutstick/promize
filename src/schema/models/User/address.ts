interface IAddress {
    address: string;
    city: string;
    country: string;
    zip: string;
}

const Address = {
    address: String,
    city: String,
    country: String,
    zip: String,
};

export { IAddress, Address };
