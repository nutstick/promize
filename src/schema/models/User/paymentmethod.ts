interface IPaymentMethod {
    credit_card_number: string;
    valid_from_month: number;
    valid_from_year: number;
}

const PaymentMethod = {
    credit_card_number: String,
    valid_from_month: Number,
    valid_from_year: Number,
};

export { IPaymentMethod, PaymentMethod };
