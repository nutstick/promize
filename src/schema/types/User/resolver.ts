import { IResolver } from '../index';

const resolver: IResolver<any, any> = {
  User: {
  },
};

export default resolver;
