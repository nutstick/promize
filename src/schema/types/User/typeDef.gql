# Account data type
type Account {
  # Email
  email: String
  # Facebook access code
  facebookAccessCode: String
  # Google access code
  googleAccessCode: String
}

# Address type
type Address implements Node {
  _id: ID!
  # Address detail
  address: String!
  # City
  city: String!
  # Country
  country: String!
  # Zip code
  zip: String!
}

# Payment Method type
type PaymentMethod implements Node {
  _id: ID!
  # Credit card number
  creditCardNumber: String!
}

enum GenderEnum {
  male
  female
}

enum CoSellerRegisterStatus {
  IDLE
  PENDING
  APPROVED
}

# User interface
interface UserType {
  _id: ID!
  # First name
  firstName: String!
  # Middle name
  middleName: String
  # Last name
  lastName: String!
  # Gender
  gender: GenderEnum!
  # Telephone number
  telNumber: String
  # Account data
  account: Account
  # Addresses
  address(id: ID): Address
  addresses: [Address]
  # Paymethods
  paymentMethod(id: ID): PaymentMethod
  paymentMethods: [PaymentMethod]
  # User avatar
  avatar: String

  # Order receipts
  orderReceipts(
    first: Int,
    after: ID,
    last: Int,
    before: ID,
    orderBy: [OrderReceiptOrdering],
  ): OrderReceiptPage

  traderooms: [TradeRoom]
  traderoom(id: ID!): TradeRoom

  createAt: Date!
  updateAt: Date!
}

# User type
type User implements UserType, Node {
  _id: ID!
  # First name
  firstName: String!
  # Middle name
  middleName: String
  # Last name
  lastName: String!
  # Gender
  gender: GenderEnum!
  # Telephone number
  telNumber: String
  # Account data
  account: Account
  # Addresses
  address(id: ID): Address
  addresses: [Address]
  # Paymethods
  paymentMethod(id: ID): PaymentMethod
  paymentMethods: [PaymentMethod]
  # User avatar
  avatar: String

  # Citizen card image
  citizenCardImage: String

  # Order receipts
  orderReceipts(
    first: Int,
    after: ID,
    last: Int,
    before: ID,
    orderBy: [OrderReceiptOrdering],
  ): OrderReceiptPage

  coSellerRegisterStatus: CoSellerRegisterStatus!

  traderooms: [TradeRoom]
  traderoom(id: ID!): TradeRoom

  createAt: Date!
  updateAt: Date!
}

# Co-Seller type
type CoSeller implements UserType, Node {
  _id: ID!
  # First name
  firstName: String!
  # Middle name
  middleName: String
  # Last name
  lastName: String!
  # Gender
  gender: GenderEnum!
  # Telephone number
  telNumber: String
  # Account data
  account: Account
  # Addresses
  address(id: ID): Address
  addresses: [Address]
  # Paymethods
  paymentMethod(id: ID): PaymentMethod
  paymentMethods: [PaymentMethod]
  # User avatar
  avatar: String

  # Citizen card image
  citizenCardImage: String

  # Order receipts
  orderReceipts(
    first: Int,
    after: ID,
    last: Int,
    before: ID,
    orderBy: [OrderReceiptOrdering],
  ): OrderReceiptPage
  
  # CoSeller flag
  coseller: Boolean!
  # Products's Own
  products(
    first: Int,
    after: ID,
    last: Int,
    before: ID,
    orderBy: [OrderReceiptOrdering],
  ): ProductPage!

  # Buy order receipts with status
  buyOrderReceipts(
    first: Int,
    after: ID,
    last: Int,
    before: ID,
    orderBy: [OrderReceiptOrdering],
    status: OrderReceiptStatus,
  ): OrderReceiptPage

  # Count order receipts with status
  totalBuyOrderReceipts(status: OrderReceiptStatus): Int

  traderooms: [TradeRoom]
  traderoom(id: ID!): TradeRoom

  createAt: Date!
  updateAt: Date!
}


# Admin type
type Admin implements UserType, Node {
  _id: ID!
  # First name
  firstName: String!
  # Middle name
  middleName: String
  # Last name
  lastName: String!
  # Gender
  gender: GenderEnum!
  # Telephone number
  telNumber: String
  # Account data
  account: Account
  # Addresses
  address(id: ID): Address
  addresses: [Address]
  # Paymethods
  paymentMethod(id: ID): PaymentMethod
  paymentMethods: [PaymentMethod]
  # User avatar
  avatar: String

  traderooms: [TradeRoom]
  traderoom(id: ID!): TradeRoom

  # Order receipts
  orderReceipts(
    first: Int,
    after: ID,
    last: Int,
    before: ID,
    orderBy: [OrderReceiptOrdering],
  ): OrderReceiptPage
    
  traderooms: [TradeRoom]
  traderoom(id: ID!): TradeRoom

  admin: Boolean
  # Pending CoSellers
  pendingCoSellers: [User]

  createAt: Date!
  updateAt: Date!
}
