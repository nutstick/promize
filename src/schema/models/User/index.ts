import * as Iridium from 'iridium';
import { Collection, Index, Instance, ObjectID, Property } from 'iridium';
import { Account, IAccountDocument } from './account';
import { Address, IAddress } from './address';
import { IPaymentMethod, PaymentMethod } from './paymentmethod';

interface IUserDocument {
  _id?: string;
  first_name: string;
  middle_name?: string;
  last_name: string;
  tel_number?: string;
  account: IAccountDocument;
  addresses?: IAddress[];
  payment_methods?: IPaymentMethod[];
  avatar: string;
  createAt?: Date;
  updateAt?: Date;
}

@Index({
  'account.email': 1,
  'account.facebookAccessCode': 1,
  'account.googleAccessCode': 1,
}, { unique: true })
@Collection('users')
class User extends Instance<IUserDocument, User> implements IUserDocument {
  @ObjectID
  _id: string;
  @Property(/^.+$/, true)
  first_name: string;
  @Property(/^.+$/, false)
  middle_name: string;
  @Property(/^.+$/, true)
  last_name: string;

  @Property(/^[0-9-]+$/, false)
  tel_number: string;

  @Property(Account, true)
  account: IAccountDocument;
  @Property(Address, false)
  address: IAddress;
  @Property(PaymentMethod, false)
  payment_method: IPaymentMethod;

  @Property(String, true)
  avatar: string;
  @Property(Date, false)
  createAt: Date;
  @Property(Date, false)
  updateAt: Date;

  static onCreating(user: IUserDocument) {
    user.addresses = [];
    user.payment_methods = [];
    user.createAt = new Date();
    user.updateAt = new Date();

    if (!user.account.facebook && !user.account.password) {
      return Promise.reject(new Error('expected one login method'));
    }
  }

  static onSaving(user: User, changes: Iridium.Changes) {
    user.updateAt = new Date();
  }
}

export { IUserDocument, User };
