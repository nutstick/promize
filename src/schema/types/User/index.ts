import { IAddress } from '../../models/User/address';
import { IPaymentMethod } from '../../models/User/paymentmethod';
import { IOrderReceipt } from '../OrderReceipt';
import { INode, IPage } from '../Pagination';
import { IProduct } from '../Product/index';
import resolver from './resolver';
import * as type from './typeDef.gql';

interface IAccount {
  email: string;
}

interface IUserType {
  firstName?: string;
  middleName?: string;
  lastName?: string;
  telNumber: string;
  account?: IAccount;
  addresses?: IAddress[];
  paymentMethods?: IPaymentMethod[];
  avatar?: string;
  orderReceipts?: IOrderReceipt[];
  createAt?: Date;
  updateAt?: Date;
}

interface IUser extends IUserType, INode {}

interface ICoSeller extends IUserType, INode {
  products?: IPage<IProduct>;
  buyOrderReceipts?: IOrderReceipt[];
}

export {
  resolver,
  type,
  IUser,
  ICoSeller,
};
