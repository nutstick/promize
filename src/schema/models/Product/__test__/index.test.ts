import { Core, Model, toObjectID } from 'iridium';
import { IProductDocument, Product } from '../';
import { mongodb } from '../../../../config';
import { IUserDocument, User } from '../../User';

// tslint:disable-next-line:no-var-requires
const m = require('casual');

const core = new Core({
  ...mongodb,
  database: 'test',
});

const model = new Model<IProductDocument, Product>(core, Product);
const userModel = new Model<IUserDocument, User>(core, User);

// Pass generator as callback
const array_of = (times, generator) => {
    return Array.apply(null, Array(times)).map(() => generator());
};

const start_date = m.moment;
const end_date = start_date.add(Math.floor(Math.floor(Math.random() * 60)), 'days');

beforeAll(async () => {
  await core.connect();
  await core.connection.dropDatabase();

  // Fixed new Date
  const FIXED_DATE = new Date('2017-06-13T04:41:20');
  (global as any).Date = jest.fn(() => FIXED_DATE);
});

afterAll(() => core.close());

describe('Product Model', () => {
<<<<<<< HEAD
  it('Product create correctly', async () => {
    const user = await userModel.create({
      _id: '5a10494ee67657f236e6b0a2',
      first_name: m.first_name,
      last_name: m.last_name,
      tel_number: m.phone,
      account: {
          email: m.email,
          password: m.password,
      },
      avatar: m.url,
    });
=======
    it('Product create correctly', async () => {
        const user = await user_model.create({
            first_name: m.first_name,
            last_name: m.last_name,
            tel_number: m.phone,
            account: {
                email: m.email,
                password: m.password,
            },
            avatar: m.url,
        });

        const product = await model.create({
            name: m.word,
            description: m.description,
            type: 'BuyNowProduct',
            // price: m.integer(100, 2000),

            picture: array_of(Math.floor(Math.random() * 3) + 1, () => m.random_element([
                'https://th-live-02.slatic.net/p/7/hequ-1483111676-123106' +
                '5-c566b543a82cfe5a0e279dbf161bd13e-catalog_233.jpg',
            ])),
            hashtag: array_of(Math.floor(Math.random() * 3), () => m.random_element([
                'uniqlo', 'HandM', 'AIIZ', 'GAP', 'Crocs', 'anello', 'kanken',
            ])),
            colors: array_of(Math.floor(Math.random() * 3), () => m.color_name),
            sizes: array_of(Math.floor(Math.random() * 3), () => m.random_element(['S', 'M', 'L', 'XL', 'XXL'])),

            promotion_start: m.moment.toDate(),
            promotion_end: m.moment.add(Math.floor(Math.floor(Math.random() * 60)), 'days').toDate(),
            owner: user._id,
        });
>>>>>>> bca2a439d522af2f8a3462811acaa7e9c2a9575e

    await model.create({
      _id: '585b11e7adb8b5f2d655da01',
      name: 'Zara',
      description: 'Zara clothes',
      original_price: 1000,
      type: 'BuyNowProduct',
      price: 800,
      pictures: [
          'https://th-live-02.slatic.net/p/7/hequ-1483111676-123106' +
          '5-c566b543a82cfe5a0e279dbf161bd13e-catalog_233.jpg',
      ],
      hashtags: ['uniqlo', 'HandM', 'AIIZ', 'GAP', 'Crocs', 'anello', 'kanken'],
      colors: ['red', 'blue'],
      sizes: ['S', 'M', 'L', 'XL', 'XXL'],
      promotion_start: new Date(2017, 13, 1, 12),
      promotion_end: new Date(2017, 14, 1, 19),
      owner: user._id,
    });

    // Number of product in database should be 1.
    const count = await model.find().count;
    expect(count).toMatchSnapshot();
    // A product should create in database..
    const products = await model.find().toArray();
    expect(products).toMatchSnapshot();
    // Should be able to find a product that has been created.
    const product = await model.findOne({ name: 'Zara' });
    expect(product).toMatchSnapshot();
  });
});
