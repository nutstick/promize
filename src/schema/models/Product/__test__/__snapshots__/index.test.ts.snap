// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Model Product create correctly 1`] = `[Function]`;

exports[`Product Model Product create correctly 2`] = `
Array [
  Object {
    "_id": "585b11e7adb8b5f2d655da01",
    "colors": Array [
      "red",
      "blue",
    ],
    "createAt": 2017-06-12T21:41:20.000Z,
    "description": "Zara clothes",
    "hashtags": Array [
      "uniqlo",
      "HandM",
      "AIIZ",
      "GAP",
      "Crocs",
      "anello",
      "kanken",
    ],
    "name": "Zara",
    "original_price": 1000,
    "owner": "5a10494ee67657f236e6b0a2",
    "pictures": Array [
      "https://th-live-02.slatic.net/p/7/hequ-1483111676-1231065-c566b543a82cfe5a0e279dbf161bd13e-catalog_233.jpg",
    ],
    "price": 800,
    "promotion_end": 2017-06-12T21:41:20.000Z,
    "promotion_start": 2017-06-12T21:41:20.000Z,
    "sizes": Array [
      "S",
      "M",
      "L",
      "XL",
      "XXL",
    ],
    "type": "BuyNowProduct",
    "updateAt": 2017-06-12T21:41:20.000Z,
  },
]
`;

exports[`Product Model Product create correctly 3`] = `
Object {
  "_id": "585b11e7adb8b5f2d655da01",
  "colors": Array [
    "red",
    "blue",
  ],
  "createAt": 2017-06-12T21:41:20.000Z,
  "description": "Zara clothes",
  "hashtags": Array [
    "uniqlo",
    "HandM",
    "AIIZ",
    "GAP",
    "Crocs",
    "anello",
    "kanken",
  ],
  "name": "Zara",
  "original_price": 1000,
  "owner": "5a10494ee67657f236e6b0a2",
  "pictures": Array [
    "https://th-live-02.slatic.net/p/7/hequ-1483111676-1231065-c566b543a82cfe5a0e279dbf161bd13e-catalog_233.jpg",
  ],
  "price": 800,
  "promotion_end": 2017-06-12T21:41:20.000Z,
  "promotion_start": 2017-06-12T21:41:20.000Z,
  "sizes": Array [
    "S",
    "M",
    "L",
    "XL",
    "XXL",
  ],
  "type": "BuyNowProduct",
  "updateAt": 2017-06-12T21:41:20.000Z,
}
`;
